doc/man3/EVP_PKEY_set1_RSA.pod

```diff
@@ -62,13 +62,16 @@ see L<openssl_user_macros(7)>:
 EVP_PKEY_get_base_id() returns the type of I<pkey>. For example
 an RSA key will return B<EVP_PKEY_RSA>.
 
-EVP_PKEY_get_id() returns the actual OID associated with I<pkey>.
-Historically keys using the same algorithm could use different OIDs.
-For example an RSA key could use the OIDs corresponding to
+EVP_PKEY_get_id() returns the actual NID associated with I<pkey>
+if I<pkey> is not implemented in an external L<provider(7)>.
```

Comment:
This isn't quite true...  it can return a proper NID for the algorithm if there's also a legacy implementation, like those built-in in libcrypto, or an ENGINE implementation.  `EVP_PKEY_KEYMGMT` will be returned if the algorithm is *only* implemented in a provider.  I'm not sure how to make that into a nicely phrased negative, unfortunately...

Reply:
Hmm -- I don't see the contradiction:

You write: 

> EVP_PKEY_KEYMGMT will be returned if the algorithm is only implemented in a provider. 

the proposal reads

> [positive NID will be returned] if I<pkey> is not implemented in an external L<provider

Sounds pretty much identical to me (just without going into detail as to what is "legacy" or engine-based) as that doesn't really matter (or needs to interest users) here, right?

Reply:
There's a fine difference in logical implication....  but you know what, I think you're right that this might be a bit too detailed to digest, i.e. can be called "implementation details" and left alone.  So I'll leave this alone.